front/src/components/MapComponent.vue

```diff
@@ -1,52 +1,84 @@
 <script setup lang="ts">
 import mapboxgl from "mapbox-gl";
+import type * as Geojson from "geojson";
 import { onMounted, ref, watch } from "vue";
 import { useRoute, useRouter } from "vue-router";
 import { useMapPointsStore } from "@/stores/mapPoints";
 
-const router = useRouter();
-const route = useRoute();
-const MAPBOX_ACCESS_TOKEN = import.meta.env.VITE_MAPBOX_ACCESS_TOKEN;
-mapboxgl.accessToken = MAPBOX_ACCESS_TOKEN;
+import type { Point } from "@/stores/mapPoints";
+export interface DirectionResponse {
+  waypoints: Waypoint[];
+  routes: Route[];
+  code: string;
+}
 
-interface MapProps {
-  center: [number, number];
-  maxBounds: [[number, number], [number, number]];
+export interface Route {
```

Comment:
Put all your interface on the directory interface and in the file main.ts
(For centralize)
And call it with import type

Reply:
For each interface or type